packages/fluentui/react/src/components/Tree/Tree.tsx

```diff
@@ -225,101 +186,102 @@ class Tree extends AutoControlledComponent<WithAsProp<TreeProps>, TreeState> {
+      activeItemIds.push(id);
+    }
 
-      siblings.forEach(sibling => {
-        if (hasSubtree(sibling) && !this.isActiveItem(sibling['id'])) {
-          activeItemIds.push(sibling['id']);
-        }
-      });
+    this.setActiveItemIds(e, activeItemIds);
+  };
+  onFocusFirstChild = currentElement => {
+    if (!currentElement || !currentElement.current) {
+      return;
+    }
 
-      if (hasSubtree(treeItemProps) && !this.isActiveItem(id)) {
-        activeItemIds.push(id);
-      }
+    const elementToBeFocused = getNextElement(this.treeRef.current, currentElement.current);
 
-      this.setActiveItemIds(e, activeItemIds);
+    if (!elementToBeFocused) {
+      return;
+    }
 
-      _.invoke(predefinedProps, 'onSiblingsExpand', e, treeItemProps);
+    elementToBeFocused.focus();
+  };
+  onFocusParent = parentRef => {
+    if (!parentRef || !parentRef.current) {
+      return;
     }
-  });
+
+    parentRef.current.focus();
+  };
 
   setActiveItemIds = (e: React.SyntheticEvent, activeItemIds: string[]) => {
-    _.invoke(this.props, 'onActiveItemIds', e, { ...this.props, activeItemIds });
+    _.invoke(this.props, 'onActiveItemIdsChange', e, { ...this.props, activeItemIds });
 
     this.setState({
       activeItemIds
     });
   };
 
   renderContent(accessibility: ReactAccessibilityBehavior): React.ReactElement[] {
-    const { itemsForRender } = this.state;
     const { items, renderItemTitle } = this.props;
 
     if (!items) return null;
 
-    const renderItems = (items: ShorthandCollection<TreeItemProps>): React.ReactElement[] => {
-      return items.reduce((renderedItems: React.ReactElement[], item: ShorthandValue<TreeItemProps>) => {
-        const itemForRender = itemsForRender[item['id']];
-        const { elementRef, ...restItemForRender } = itemForRender;
+    const renderItems = (
+      items: ShorthandCollection<TreeItemProps>,
+      level = 1,
+      parentRef?: React.RefObject<HTMLElement>
+    ): React.ReactElement[] => {
+      return items.reduce((renderedItems: React.ReactElement[], item: ShorthandValue<TreeItemProps>, index: number) => {
         const isSubtree = hasSubtree(item);
         const isSubtreeExpanded = isSubtree && this.isActiveItem(item['id']);
-        const renderedItem = TreeItem.create(item, {
-          defaultProps: () => ({
-            accessibility: accessibility.childBehaviors ? accessibility.childBehaviors.item : undefined,
-            className: Tree.slotClassNames.item,
-            expanded: isSubtreeExpanded,
-            renderItemTitle,
-            key: item['id'],
-            contentRef: elementRef,
-            ...restItemForRender
-          }),
-          overrideProps: this.handleTreeItemOverrides
-        });
-
-        return [...renderedItems, renderedItem, ...(isSubtreeExpanded ? renderItems(item['items']) : ([] as any))];
+        const elementRef = React.createRef<HTMLElement>();
+        const contextValue: TreeItemRenderContextValue = {
```

Comment:
This will not work 😨 https://smykhailov.github.io/react-patterns/#/context

So, even if consumers will inject `React.memo()` around `TreeItem` they will still receive updates as context value will always change.